packages/compass-preferences-model/src/react.ts

```diff
@@ -0,0 +1,50 @@
+import preferencesAccess from './';
+import type { AllPreferences } from './';
+
+interface ReactHooks {
+  useState: <T>(initialValue: T) => [T, (newValue: T) => void];
+  useEffect: (
+    effectFn: () => () => void,
+    dependencies: readonly unknown[]
+  ) => void;
+}
+
+/** Use as: const enableMaps = usePreference('enableMaps', React); */
+export function usePreference<K extends keyof AllPreferences>(
+  key: K,
+  { useState, useEffect }: ReactHooks
```

Comment:
Tiny nit, feel free to ignore: I'd probably still suggest not to inject react here and just have it as a dependency and use directly as this is kinda uncommon to inject React like that and pretty much nothing else can be passed here except for React and from my perspective not having react in direct deps is less beneficial in this case than smaller interface

Reply:
Mh yeah, I'm not sure it's worth the simplification. I understand that practically speaking, we don't have to worry right now about this working with different React versions or similar scenarios (I don't think it would make sense to pass in anything else other than a React import, yeah), but I do think it's fine to keep the code generic and just take what we need from the library. The tests here should ensure that passing in React works both from on the type and the runtime levels.